cid.go

```diff
@@ -522,6 +525,74 @@ func (c Cid) Prefix() Prefix {
 	}
 }
 
+func (c Cid) MarshalCBOR(w io.Writer) error {
+	tag := cbg.CborEncodeMajorType(6, 42)
```

Comment:
If we're going to be _manually_ parsing CBOR, let's at least get these into constants.

Reply:
Ideally, we'd have one function per major type:

* WriteTag
* WriteInt
* ReadTag
* ReadInt
* WriteBytes

...